### DISCUSSION TITLE
Searching multiple collections for higher recall

### DISCUSSION THREAD

**Discussion by dutysloth** [milvus-io/milvus#31083] [Q&A and General discussion]
Hi,
I just thought of an idea to cross-check multiple collections with different index types to increase recall rate, and wanted to hear your opinions about it.

Explanation:
Lets say I have a collection A with IVF_FLAT index that gives me 99% recall rate
And another collection B with HNSW that gives also 99% recall rate

Now when I search a given vector X in collection A there is a chance that it will fall into the 1% and give wrong results.
Of course I can play with parameters such as NPROBE to increase recall rate.

But if I also search collection B and then compose my top-K list in the following method:
Top1(collectionA), Top1(CollectionB), Top2(CollectionA), Top2(CollectionB), Top3(CollectionA),...
(and remove duplicates)
The chances that the given vector X will fall into the 1% in both collections is very slim because they use different index types (I assume).

Obviously the storage and resources cost will increase because of the data duplication, but other methods (e.g. increasing NPROBE) will also cause that.

Am I correct in my thinking? Is cross-checking a common thing or maybe someone tried it already?

**Comment by xiaofan-luan** [comment]
usually a smart idea to improve recall is simply try to search more data and try to do oversampling.
If you want to get top10 result, you can retrieve top 50 do another refine.

As you said, search on multiple index is another way to do oversampling. It brings you extra accuracy with more memory and search cost.

In milvus 2.4 we will support multivector column and you can have multivectors search and rank them with RRF or Weighted strategy
